pkg/migration/controllers/migration.go

```diff
@@ -1499,10 +1513,120 @@ func (m *MigrationController) getParsedLabels(
 	return m.getParsedMap(labels, clusterPair)
 }
 
+// updateOwnerReferenceOnPVC updates owner reference on PVC objects
+// on destination cluster
+func (m *MigrationController) updateOwnerReferenceOnPVC(
+	migration *stork_api.Migration,
+	objects []runtime.Unstructured,
+	clusterPair *stork_api.ClusterPair,
+	pvcsWithOwnerRef []v1.PersistentVolumeClaim,
+) error {
+	remoteConfig, err := getClusterPairSchedulerConfig(migration.Spec.ClusterPair, migration.Namespace)
+	if err != nil {
+		return err
+	}
+	remoteAdminConfig := remoteConfig
+	// Use the admin cluter pair for cluster scoped resources if it has been configured
+	if migration.Spec.AdminClusterPair != "" {
+		remoteAdminConfig, err = getClusterPairSchedulerConfig(migration.Spec.AdminClusterPair, m.migrationAdminNamespace)
+		if err != nil {
+			return err
+		}
+	}
+	adminClient, err := kubernetes.NewForConfig(remoteAdminConfig)
+	if err != nil {
+		return err
+	}
+	remoteInterface, err := dynamic.NewForConfig(remoteConfig)
+	if err != nil {
+		return err
+	}
+	remoteAdminInterface := remoteInterface
+	if migration.Spec.AdminClusterPair != "" {
+		remoteAdminInterface, err = dynamic.NewForConfig(remoteAdminConfig)
+		if err != nil {
+			return err
+		}
+	}
+
+	ruleset := m.getDefaultRuleSet()
+	for _, srcPvc := range pvcsWithOwnerRef {
+		destPvc, err := adminClient.CoreV1().PersistentVolumeClaims(srcPvc.GetNamespace()).Get(context.TODO(), srcPvc.Name, metav1.GetOptions{})
+		if err != nil {
+			return err
+		}
+		// Skip updating if destination PVC has owner reference enabled
+		if len(destPvc.GetOwnerReferences()) > 0 {
```

Comment:
Why do we need this check? I think we should keep making sure that the PVC on destination side always points to the right owner.

Reply:
Agree. removing this check.